### PR TITLE
Update v3 SSP compsets/configurations/tests and use_case files

### PR DESCRIPTION
**Pull request by wlin7**
Add/update SSP compsets with standard v3 configurations. The changes include:

- SSP245, SSP370, SSP585
- atm and land configurations
- atm and land use_case files
- wcprodssp testmod (replacing workaround that are based on hybrid v2/v3 configurations)

[non-BFB] only for SSP tests
[NML] changes only for SSP tests

---------------------------------------------------------
Checklist and Notes

- E90 emission for SSP scenarios need to be inspected. The one specified was created in 2017 and uses variable emis_e90, while emis_anthro is used for piControl/historical and F compsets (updated with the one created for all v3 SSP scenarios) (done)
- E90 emission file scenario independent (confirmed)
- LULC (landuse.timeseries) file at r05 are needed for SSP245 and SSP585 (done, note that LULC files are independent of land/ocean mask)
- Volcanic SO2 emission file for SSP245 needs to be relocated (currently put on the directory for SSP370 forcings) (done)
- DMS flux file scenario independent?
- fsurdat, 1850 or 2015 more appropriate? If 2015, could be made scenario dependent.
- use new prescribed oxidant file oxid_SSP370_1.9x2.5_L70_1849-2101_c20240228.nc (replacing oxid_SSP370_1.9x2.5_L70_2015-2100_c20211006.nc)
- Also use oxidant files c20240228 for SSP245 and SSP585
- linv3 file, which version to use, c20230705 or c20231207? piControl and historical have been using the c20230705 version.
- SSP370, SSP245, SSP585 as is run ok.
- SS
Available test runs for comparing nml files (atm_in and lnd_in) if they are helpful for your review
- historical: /lcrc/group/e3sm2/ac.wlin/E3SMv3/v3.LR.historical_0051/run
- SSP245: /lcrc/group/e3sm2/ac.wlin/E3SMv3/SSP/v3.LR.ssp245_0051/tests/S_1x2_nmonths/run
- SSP370: /lcrc/group/e3sm2/ac.wlin/E3SMv3/SSP/v3.LR.ssp370_0051/tests/S_1x2_nmonths/run
- SSP585: /lcrc/group/e3sm2/ac.wlin/E3SMv3/SSP/v3.LR.ssp585_0051/tests/S_1x2_nmonths/run

### CODE REVIEW

**Comment by tangq** [comment]
Hi @wlin7 , regarding your E90 checklist and notes below,

I assume the file created in 2017 refers to this file `/lcrc/group/e3sm/data/inputdata/atm/cam/chem/emis/CMIP6_emissions_1750_2015/emissions_E90global_surface_1750-2100_0.9x1.25_c20170322.nc`, which was before E90 was implemented for E3SMv3. We certainly do NOT want to use this file in E3SM.

For E3SMv3, we should use this E90 file for ALL SSP scenarios because E90 is an artificial tracer which doesn't change with SSPs. 
`/lcrc/group/acme/public_html/inputdata/atm/cam/chem/trop_mozart/ub/emissions_E90_surface_1750-2101_1.9x2.5_c20231222.nc`, which was created by Jinbo a couple months ago. (@jinboxie can provide details if needed.)

> Checklist and Notes

> E90 emission for SSP scenarios need to be inspected. The one specified was created in 2017 and uses variable emis_e90,  while emis_anthro is used for piControl/historical and F compsets

**Comment by wlin7** [comment]
Thank you @tangq and @jinboxie . It is great you have already created one for all v3 SSP scenarios.

**Review by keziming — APPROVED** [review]

File: components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+<!-- <tracer_cnst_filelist>this_field_is_not_used</tracer_cnst_filelist> -->
+
+<!-- prescribed methane  -->
+<prescribed_ghg_file      >ch4_oxid_1.9x2.5_L26_1990-1999clim.c090804.nc</prescribed_ghg_file>
+<prescribed_ghg_datapath  >atm/cam/chem/methane</prescribed_ghg_datapath>
+<prescribed_ghg_type      >CYCLICAL</prescribed_ghg_type>
+<prescribed_ghg_cycle_yr  >1995</prescribed_ghg_cycle_yr>
+<prescribed_ghg_filelist  >''</prescribed_ghg_filelist>
+<prescribed_ghg_specifier>'prsd_ch4:CH4'</prescribed_ghg_specifier>
```
Comment (tangq) [inline components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:75]: I am not sure why we need this prescribed methane file, which appears to be the seasonal cycle climatology of 1990-1999, for the SSPs. I confirmed that "prsd_ch4" is not defined in chem_mech.in, so it is highly likely these settings are not used in the simulation. Shall we delete it here and in other two SSP use case files too?

File: components/eam/bld/namelist_files/use_cases/SSP370_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml [hunk 2]
```diff
@@ -38,24 +32,60 @@
...
 
 <!-- <tracer_cnst_filelist>this_field_is_not_used</tracer_cnst_filelist> -->
 
+<!-- prescribed methane  -->
+<prescribed_ghg_file      >ch4_oxid_1.9x2.5_L26_1990-1999clim.c090804.nc</prescribed_ghg_file>
+<prescribed_ghg_datapath  >atm/cam/chem/methane</prescribed_ghg_datapath>
+<prescribed_ghg_type      >CYCLICAL</prescribed_ghg_type>
+<prescribed_ghg_cycle_yr  >1995</prescribed_ghg_cycle_yr>
+<prescribed_ghg_filelist  >''</prescribed_ghg_filelist>
+<prescribed_ghg_specifier>'prsd_ch4:CH4'</prescribed_ghg_specifier>
```
Comment (tangq) [inline components/eam/bld/namelist_files/use_cases/SSP370_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:75]: See comments above.

File: components/eam/bld/namelist_files/use_cases/SSP585_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml [hunk 3]
```diff
@@ -38,24 +32,60 @@
...
 
 <!-- <tracer_cnst_filelist>this_field_is_not_used</tracer_cnst_filelist> -->
 
+<!-- prescribed methane  -->
+<prescribed_ghg_file      >ch4_oxid_1.9x2.5_L26_1990-1999clim.c090804.nc</prescribed_ghg_file>
+<prescribed_ghg_datapath  >atm/cam/chem/methane</prescribed_ghg_datapath>
+<prescribed_ghg_type      >CYCLICAL</prescribed_ghg_type>
+<prescribed_ghg_cycle_yr  >1995</prescribed_ghg_cycle_yr>
+<prescribed_ghg_filelist  >''</prescribed_ghg_filelist>
+<prescribed_ghg_specifier>'prsd_ch4:CH4'</prescribed_ghg_specifier>
```
Comment (tangq) [inline components/eam/bld/namelist_files/use_cases/SSP585_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:75]: see comments above.

**Review by tangq — APPROVED** [review]
The PR looks good to me except the prescribed methane settings pointing to the 1990s climatology. These settings are likely not used and should be deleted to avoid confusions.

File: components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml (hunk 1 above)
Comment (wlin7) [reply components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:75]: Nice catch, @tangq . We did have discussion on this (#5781, #5823). It is not needed.

**Review by mingxuanwupnnl — APPROVED** [review]
Thank you @wlin7  Emission files and oxidant files look good to me.

File: components/elm/bld/namelist_files/use_cases/2015-2100_SSP245_transient.xml [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+<model_year_align_ndep  phys="elm" use_cn=".true."   >1850</model_year_align_ndep>
+
+<stream_year_first_pdep phys="elm" use_cn=".true."   >2000</stream_year_first_pdep>
+<stream_year_last_pdep  phys="elm" use_cn=".true."   >2000</stream_year_last_pdep>
+<model_year_align_pdep  phys="elm" use_cn=".true."   >2000</model_year_align_pdep>
+
+<stream_year_first_popdens phys="elm" use_cn=".true."   >1850</stream_year_first_popdens>
+<stream_year_last_popdens  phys="elm" use_cn=".true."   >2010</stream_year_last_popdens>
+<model_year_align_popdens  phys="elm" use_cn=".true."   >1850</model_year_align_popdens>
+-->
```
Comment (bishtgautam) [inline components/elm/bld/namelist_files/use_cases/2015-2100_SSP245_transient.xml:30]: These might need to be uncommented, and @acme-y9s might know more about these BGC settings.

**Comment by wlin7** [comment]
Hi @tangq , the settings for the prescribed methane section has been all removed from the SSP use_case files via 
9a0ab28 . Note that similar settings still exist in other use_case _files (e.g., 1850, 2010, 20TR).  Related settings still show up in atm_in. Complete removal requires removal from namelist_defaults_eam.xml and changes of logics in build-namelist (and probably configure). This is not done.

File: components/elm/bld/namelist_files/use_cases/2015-2100_SSP245_transient.xml (hunk 4 above)
Comment (wlin7) [reply components/elm/bld/namelist_files/use_cases/2015-2100_SSP245_transient.xml:30]: Thanks for careful review, @bishtgautam . I didn't realize the whole block was commented out.
The settings, if uncommented, would be the same as for historical (CMIP6_transient). Are they suitable for SSPs?

Currently, with the blocked commented out, stream_year values for `*_pdep, *_ndep, and *_popdens` are all using 2000.

File: cime_config/allactive/config_compsets.xml [hunk 5]
```diff
@@ -113,12 +113,17 @@
 
 <compset>
   <alias>WCYCLSSP585</alias>
-  <lname>SSP585SOI_EAM%CMIP6_ELM%SPBC_MPASSI_MPASO_MOSART_SGLC_SWAV</lname>
+  <lname>SSP585SOI_EAM%CMIP6_ELM%CNPRDCTCBCTOP_MPASSI_MPASO_MOSART_SGLC_SWAV</lname>
 </compset>
```
Comment (hwangacme) [inline cime_config/allactive/config_compsets.xml:117]: Is there any particular consideration to name these SSPs under WCYCL? These compsets are now being created for different groups.

File: components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+<history_aerosol>.true.</history_aerosol>
+<history_aero_optics>.true.</history_aero_optics>
+
+<!-- External forcing for BAM or MAM.  CMIP6 input4mips data -->
+<ext_frc_type>INTERP_MISSING_MONTHS</ext_frc_type>
+<no2_ext_file         >atm/cam/chem/trop_mozart_aero/emis/CMIP6_SSP245_ne30/emissions-cmip6_ssp245_e3sm_NO2_aircraft_vertical_2015-2100_1.9x2.5_c20240219.nc </no2_ext_file>
+<so2_ext_file         >atm/cam/chem/trop_mozart_aero/emis/CMIP6_SSP245_ne30/cmip6_ssp245_mam4_so2_volc_elev_2015-2100_c240331.nc </so2_ext_file>
+<soag0_ext_file       >atm/cam/chem/trop_mozart_aero/emis/CMIP6_SSP245_ne30/emissions-cmip6_ssp245_e3sm_SOAG0_elev_2015-2100_1.9x2.5_c20240219.nc </soag0_ext_file>
+<bc_a4_ext_file       >atm/cam/chem/trop_mozart_aero/emis/CMIP6_SSP245_ne30/cmip6_ssp245_mam4_bc_a4_elev_2015-2100_c200716.nc </bc_a4_ext_file>
+<mam7_num_a1_ext_file >atm/cam/chem/trop_mozart_aero/emis/CMIP6_SSP245_ne30/cmip6_ssp245_mam4_num_a1_elev_2015-2100_c200716.nc </mam7_num_a1_ext_file>
```
Comment (hwangacme) [inline components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:26]: Shall we change these "mam7" pointers of input files to "mam4"? They look quite confusing here.

**Review by hwangacme — APPROVED** [review]
No problems found. Just identified some minor naming issues.

File: components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml (hunk 6 above)
Comment (wlin7) [reply components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:26]: Indeed appear confusing. Does 'a1' refer to the same thing in mam7 and mam4?

Comment (hwangacme) [reply components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:26]: num_a1 does refer to the same thing for both MAM4 and MAM7, but num_a3 for MAM7 below refers to num_a4 for MAM4.

Comment (wlin7) [reply components/eam/bld/namelist_files/use_cases/SSP245_eam_CMIP6_chemUCI-Linoz-mam5-vbs.xml:26]: Thanks, @hwangacme .  Indeed in all the use_case files, there is a comment that mam7_num_a3 is to set num_a4 emission. If to change the pointers, need to update all use_case files and namelist_default, as well as build-namelist. These should be fine; but would also introduce NML diff for all standard tests (and make nml in ongoing production simulations look different). We can apply the changes at a later time.

File: cime_config/allactive/config_compsets.xml (hunk 5 above)
Comment (wlin7) [reply cime_config/allactive/config_compsets.xml:117]: @hwangacme , that is more of continuing the naming convention we have been having. We are still a distinction among WCYCLSSP, CRYOSSP, BGCSSP.

**Comment by rljacob** [comment]
@wlin7 is this ready for merging?

**Comment by wlin7** [comment]
@rljacob , the land settings need further review/input. @acme-y9s and @thorntonpe , can you please take a look? It is about the [use_case settings ](https://github.com/E3SM-Project/E3SM/pull/6335#discussion_r1566686744)to work with the BGC land.

File: components/elm/bld/namelist_files/use_cases/2015-2100_SSP245_transient.xml (hunk 4 above)
Comment (wlin7) [reply components/elm/bld/namelist_files/use_cases/2015-2100_SSP245_transient.xml]: Going with the transient settings as for the historical use_case for now.

**Comment by acme-y9s** [comment]
@wlin7 @bishtgautam For the nitrogen deposition (nden) and population density (popdens) files, we do have the transient files that we used for BGCv1 simulation (like SSP585). For the phosphorus deposition file (pden) we only one year. If you do not specify the those files in your use_case file, the model will have default setting files (maybe based on you simulation year ), but the default files may not what you need. There should be ndep and popdens files for each SSPs, you can specify thoses files in each SSP's use_case file. I did specify those files for our BGCv1 SSP585 case, you can have a look if you want. Or, if you need help, feel free to let me know.

File: cime_config/testmods_dirs/allactive/wcprodssp/user_nl_elm [hunk 7]
```diff
@@ -2,13 +2,45 @@
...
+               'SOIL3N_TNDNCY_VERT_TRANS','SOIL3N_TO_SOIL4N','SOIL3N_vr','SOIL3P_TNDNCY_VERT_TRANS','SOIL3P_TO_SOIL4P',
+               'SOIL3P_vr','SOIL4_HR','SOIL4C_vr','SOIL4N_TNDNCY_VERT_TRANS','SOIL4N_TO_SMINN',
+               'SOIL4N_vr','SOIL4P_TNDNCY_VERT_TRANS','SOIL4P_TO_SMINP','SOIL4P_vr','SOLUTIONP_vr',
+               'TCS_MONTH_BEGIN','TCS_MONTH_END','TOTCOLCH4','water_scalar','WF',
+               'wlim_m','WOODC_LOSS','WTGQ'
+ hist_fincl1 = 'SNOWDP','COL_FIRE_CLOSS','NPOOL','PPOOL','TOTPRODC'
  hist_fincl2 = 'H2OSNO', 'FSNO', 'QRUNOFF', 'QSNOMELT', 'FSNO_EFF', 'SNORDSL', 'SNOW', 'FSDS', 'FSR', 'FLDS', 'FIRE', 'FIRA'
  hist_mfilt = 1,365
- hist_nhtfrq = -24,-24
+ hist_nhtfrq = 0,-24
```
Comment (rljacob) [inline cime_config/testmods_dirs/allactive/wcprodssp/user_nl_elm:44]: should be -24 for testing.

Comment (wlin7) [reply cime_config/testmods_dirs/allactive/wcprodssp/user_nl_elm:44]: Thanks for catching it. Done, for user_nl_eam as well.

**Comment by wlin7** [comment]
> @wlin7 @bishtgautam For the nitrogen deposition (nden) and population density (popdens) files, we do have the transient files that we used for BGCv1 simulation (like SSP585). For the phosphorus deposition file (pden) we only one year. If you do not specify the those files in your use_case file, the model will have default setting files (maybe based on you simulation year ), but the default files may not what you need. There should be ndep and popdens files for each SSPs, you can specify thoses files in each SSP's use_case file. I did specify those files for our BGCv1 SSP585 case, you can have a look if you want. Or, if you need help, feel free to let me know.

Thanks, @acme-y9s definitely need more help from you on this. Can you please provide the files (the path if ready on our machine) and the proper year-align settings? We can update the specification in a next PR.

**Comment by wlin7** [comment]
Merged to next.

**Comment by wlin7** [comment]
Remerged to next after restoring the config settings for 4xCO2 and 1pctCO2 compset. Pattern `_EAM%CMIP6_` can not match the CO2 compset longname, which uses `-` in place of `_` following the string `CMIP6`.

**Comment by wlin7** [comment]
Only expected SSP test diffs with this PR on next (and compy's SSP370 FAIL due to permission to download additional forcing files).

The PR is merged to master (to create a master repo with SSP compsets). Requests for bless SSP test DIFFs are submitted but before they are blessed, both next and master will see diff on Monday.

**Comment by peterdschwartz** [comment]
@wlin7  Maybe due to machine issues at the time, but this PR may still have DIFFs on pm-cpu that need to be blessed(for example https://my.cdash.org/viewTest.php?onlyfailed&buildid=2561805).

**Pull request merged**